frontend/include/chpl/util/version-info.h

```diff
@@ -71,9 +71,9 @@ bool getIsOfficialRelease();
 
 /*
   get the build version value. This may be an integer or a git-sha depending on
-  the value of official_release and developer.
+  the value of official_release flag.
```

Comment:
We already have `gitSHA`. (Which BTW I think it'd be nice to rename & I brought that up in a comment above). I think it's misleading for `getBuildVersion` to ever return a SHA at all. Instead, I think it should return an integer that is 0 in the cases it would return a SHA today. AFAIK, the only code calling it (through the wrapper in the production compiler) is here:

https://github.com/chapel-lang/chapel/blob/9fe6f7c9044b87e9ae3c90dab99fd345b2df3c95/compiler/resolution/preFold.cpp#L2085-L2091

And that code already has a conditional to not use the result if it's an official release. So, that code would be better written to call `getShaVersion`.

Reply:
I had wanted to remove `getBuildVersion`, and changing `preFold.cpp` to use the updated `getCommitHash` function allowed me to remove it completely.